pyccl/tracers.py

```diff
@@ -404,7 +393,10 @@ def add_tracer(self, cosmo, kernel=None,
         self._trc.append(_check_returned_tracer(ret))
 
     def __del__(self):
-        if hasattr(self, '_trc'):
+        # Sometimes lib is freed before some Tracers, in which case, this
+        # doesn't work.
+        # So just check that lib.cl_tracer_t_free is still a real function.
+        if hasattr(self, '_trc') and lib.cl_tracer_t_free is not None:
```

Comment:
Nice, I got bitten in the ass by this one before and couldn't understand what was happening. I'll put this in namaster too.